event/src/main/java/com/bazaarvoice/emodb/event/db/astyanax/AstyanaxEventReaderDAO.java

```diff
@@ -473,17 +476,25 @@ public void run() {
     }
 
     /** Executes a {@code RowQuery} with {@code autoPaginate(true)} repeatedly as necessary to fetch all pages. */
-    private <K, C> Iterable<Column<C>> executePaginated(final RowQuery<K, C> query) {
-        return OneTimeIterable.wrap(Iterators.concat(new AbstractIterator<Iterator<Column<C>>>() {
+    private <K, C> Iterator<Column<C>> executePaginatedStream(final RowQuery<K, C> query) {
+        return Iterators.concat(new AbstractIterator<Iterator<Column<C>>>() {
             @Override
             protected Iterator<Column<C>> computeNext() {
                 ColumnList<C> page = execute(query);
                 return !page.isEmpty() ? page.iterator() : endOfData();
             }
-        }));
+        });
     }
 
-    private <R> R execute(Execution<R> execution) {
+    /**
+     * Convenience interface to {@link #executePaginatedStream(RowQuery)} as an {@link Iterable} backed by the
+     * paginated results which can be iterated exactly once.
+     */
+     private <K, C> Iterable<Column<C>> executePaginated(final RowQuery<K, C> query) {
```

Comment:
I may be missing something, but this method seems unnecessary to me. In addition to wrapping an `Iterator` in a `OneTimeIterable` being an antipattern, all of its callers immediately unwrap the iteratble as an iterator, so they could simply call `executePaginatedStream` instead.

Reply:
I agree, it would just cause the unrelated `for` loops to have to be rewritten using iterators.  Not a big deal, but it causes code changes unrelated to this update.  I can easily change them all to iterators and remove this method.